common/config/src/lib.rs

```diff
@@ -137,12 +142,10 @@ impl TryFrom<RawConfig> for Config {
             )
         }
 
-        template_builder.user_agent(format!(
-            "{}/{} ({})",
-            env!("CARGO_PKG_NAME"),
-            env!("CARGO_PKG_VERSION"),
-            info
-        ));
+        // Safety: this is reading from extern "Rust" statics declared in main.rs or the test module
+        let (pkg_name, pkg_version) = unsafe { (PKG_NAME, PKG_VERSION) };
```

Comment:
Can we update the comment to indicate how we know this line of code is ultimately safe?

If I understand correctly this will only panic if the main binary is missing these unmangled variables. Is it possible to provide a default for that case?

Reply:
There won't be any panics, if the symbol is not defined then the linker will error out before the build completes.

The only error case would be if the symbols are exported but have the wrong type.
Given that we are the ones exporting the symbols, and any duplicate defined elsewhere in the dependecy graph will result in a linker error that shouldn't be an issue.